### ISSUE TITLE
An easier way to create Preset with normalized values

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#157]
Currently it's a bit hit or miss with 0..1 float
Possible solution: add a `makePresetFromValues` templated function that takes a tuple and check with _client.params to know if this is a valid preset.

**Comment by p0nce** [comment]
or a fbx parser?

**Comment by 0xEAB** [comment]
I'd be careful with something reading/writing FXB/FXP since it falls under the VST2 license agreement to my understanding. Which means, it would get unusable for newcomers soon.

**Comment by p0nce** [comment]
As a work-around solution for now, FXB parsing has been implemented in the generic client.
It has : B and F, but that's about it. The reasoning is that because it's not inside the generic client, but just create an array of `Preset`, it can be done inadequately. Also hosts don't export many formats.

Available in v8.4.3

Example code:
```d
    override Preset[] buildPresets()
    {
        return loadPresetsFromFXB(this, import("all-presets.fxb"));
    }    
```
This saves about 300 LOC for each plugin here!
One downside is that creating presets with code can be more resistant to parameter changes over major versions. With factory presets stored in FBX, your presets are stuck to a major version of the plugin (you can only add more parameters, not reorder or delete them, and you can't remap them).
